### ISSUE TITLE
Using a Provider as an artifact for the maven-publish-Plugin does not work in Gradle 5.6 anymore

### ISSUE THREAD

**Issue by Adrodoc** [gradle/gradle#11054]
Using a Provider<File> as an artifact for the `maven-publish`-Plugin does not work in Gradle 5.6 anymore. Using a File instance still works despite the error message (see current behaviour) suggesting otherwise.

### Expected Behavior

The file `my-artifact.txt` should be published into the local maven repository.

### Current Behavior

You get the following error:

```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':publishArtifactoryPublicationToMavenLocal'.
> Cannot convert C:\...\gradle-issue-11054\my-artifact.txt to a task.
  The following types/formats are supported:
    - A Task instance
    - A TaskReference instance
    - A Buildable instance
    - A TaskDependency instance
    - A Provider that represents a task output
    - A Provider instance that returns any of these types
    - A Closure instance that returns any of these types
    - A Callable instance that returns any of these types
    - An Iterable, Collection, Map or array instance that contains any of these types
```

### Context

This works in Gradle 5.5.1 and below, but stops working in Gradle 5.6.

### Steps to Reproduce 

Execute `gradlew publishToMavenLocal` in this project: [gradle-issue-11054.zip](https://github.com/gradle/gradle/files/3733137/gradle-issue-11054.zip)

For reference here is the `build.gradle`:

```gradle
apply plugin: 'maven-publish'

wrapper { gradleVersion = '5.6' }

group = 'org.gradle'
ext.artifactId = 'issue-11054'
version = '1.0.0'

publishing {
  publications {
    artifactory(MavenPublication) {
      artifactId = project.artifactId
      // Works in Gradle 5.5.1 and Gradle 5.6
//      artifact file('my-artifact.txt')

      // Works in Gradle 5.5.1 but not Gradle 5.6
      artifact provider{ file('my-artifact.txt') }
    }
  }
}
```

### Your Environment

```
user@PC MINGW64 /c/.../gradle-issue-11054
$ ./gradlew -v

------------------------------------------------------------
Gradle 5.6
------------------------------------------------------------

Build time:   2019-08-14 21:05:25 UTC
Revision:     f0b9d60906c7b8c42cd6c61a39ae7b74767bb012

Kotlin:       1.3.41
Groovy:       2.5.4
Ant:          Apache Ant(TM) version 1.9.14 compiled on March 12 2019
JVM:          1.8.0_162 (Oracle Corporation 25.162-b12)
OS:           Windows 7 6.1 amd64
```

**Comment by jjohannes** [comment]
Thanks for reporting this @Adrodoc. This is indeed a regression caused by the combination of #9467 and #6775 I believe. Fix is in the making.

**Comment by jjohannes** [comment]
The fix on `master` for `6.1`

Keeping this open in case there is a `6.0.1`- pick https://github.com/gradle/gradle/commit/a4b79f33fee015bb5faded27426369ad54bae8dd

**Comment by jjohannes** [comment]
Picked to `release` for `6.0.1`
